### PR TITLE
Vulkan Android Tutorial and Deploy application update + #1067 fix

### PR DESCRIPTION
**Pull request by dayanandasiet**
* vulkan rpc and java context support
* vulkan deploy support extended
* vulkan tutorial update

#1067 fix

Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from others in the community.

### CODE REVIEW

**Comment by dayanandasiet** [comment]
@yzhliu @ @PariksheetPinjari909 @janboeye 

Help to review this design and tutorial?

File: apps/android_deploy/README.md [hunk 1]
```diff
@@ -103,17 +116,8 @@ Follow instruction to get compiled version model for android target [here.](http
...
-
-`OpenCL Verison flavor`
-```
-    private static final boolean EXE_GPU            = true;
-```
-
-
 Install compiled android application on phone and enjoy the image classifier demo using extraction model
 
+Different flavor version `CPU Verison flavor` or `GPU[Vulkan/OpenCL] Version flavor` can test by choosing flavor under android application UI `TVM Android Demo->Menu->Flavor Settings`.
```
Comment (PariksheetPinjari909) [inline apps/android_deploy/README.md]: Pls correct spell 'Version'

File: apps/android_rpc/tests/android_rpc_test.py [hunk 2]
```diff
@@ -20,49 +20,78 @@
 arch = "arm64"
 target = "llvm -target=%s-linux-android" % arch
 
+test_opencl = True
+test_vulkan = False
```
Comment (PariksheetPinjari909) [inline apps/android_rpc/tests/android_rpc_test.py]: Add comment for better understanding to user

**Comment by PariksheetPinjari909** [comment]
Everything else looks good to me. Thanks @dayanandasiet

File: apps/android_deploy/app/download-models.gradle [hunk 3]
```diff
@@ -11,7 +11,7 @@
 def models = ['extraction.zip']
 
 // Root URL for model archives
-def MODEL_URL = 'https://github.com/PariksheetPinjari909/TVM_models/blob/master/extraction_model'
+def MODEL_URL = 'https://github.com/dayanandasiet/TVM_models/blob/master/darknet/extraction'
```
Comment (srkreddy1238) [inline apps/android_deploy/app/download-models.gradle]: could use dmlc/web-data for all these dependencies.

File: apps/android_deploy/app/src/main/jni/Application.mk [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+        config ?= make/config.mk
+    endif
 endif
 
 include $(config)
 
+# We target every architecture except armeabi here, for two reasons:
+# 1) armeabi is deprecated in NDK r16 and removed in r17
+# 2) vulkan is not supported in armeabi
+#APP_ABI := armeabi-v7a arm64-v8a x86 x86_64 mips
```
Comment (srkreddy1238) [inline apps/android_deploy/app/src/main/jni/Application.mk]: Pls correct these local changes.

File: docs/deploy/android.md [hunk 5]
```diff
@@ -3,23 +3,11 @@
...
-```
-lib.export_library("deploy_lib.so", ndk.create_shared)
-with open("deploy_graph.json", "w") as fo:
-    fo.write(graph.json())
-with open("deploy_param.params", "wb") as fo:
-    fo.write(nnvm.compiler.save_param_dict(params))
-```
-
-deploy_lib.so, deploy_graph.json, deploy_param.params will go to android target.
+An reference scripts can be found at [save_android_model_functions.py](https://github.com/dayanandasiet/TVM_models/blob/master/save_android_model_functions.py) which help to save deploy_lib.so, deploy_graph.json, deploy_param.params for target android phone.
```
Comment (srkreddy1238) [inline docs/deploy/android.md]: -> "A reference script"

Comment (srkreddy1238) [reply docs/deploy/android.md]: suggest to enhance "tutorials/nnvm/get_started.py" to add vulkan options section instead.
Or
@tqchen : Can we use dmlc/web-data for these utils ?

Comment (tqchen) [reply docs/deploy/android.md]: I would recommend we put common scripts under apps/android_deploy as long as it does not contain binary blobs

**Comment by tqchen** [comment]
I would like to bring your attention to https://github.com/dmlc/tvm/pull/1457, which changes the current usage of proxy to tracker, and improves the performance regression. I understand it would require a bit effort to resolve the conflict, but the new tracker mode allows us to directly connect to a fleet of mobile devices, which is a significant improved.

Please rebase and update the content accordingly

**Comment by tqchen** [comment]
@dayanandasiet can you update the PR according to comments? Thanks!

**Comment by dayanandasiet** [comment]
this PR is closed and moved with new #1488 PR .

**Pull request closed without merging**